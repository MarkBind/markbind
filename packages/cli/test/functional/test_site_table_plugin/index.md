# Welcome to MarkBind

This is to test data tables.

---
Test searchable

<d-table searchable>
| Name | Age | Country |
|------|-----|---------|
| John | 28  | USA     |
| Emily| 32  | Canada  |
| Michael | 41 | UK    |
| Sophia | 255 | Germany|
| David | 37  | Australia |
| <span><include src="others/inclusionInsideTable.md" /></span> | 29   | New Zealand |
</d-table>


---
Test sortable

<d-table sortable>
| Name | Age | Country |
|------|-----|---------|
| John | 28  | USA     |
| Emily| 32  | Canada  |
| Michael | 41 | UK    |
| Sophia | 255 | Germany|
| David | 37  | Australia |
| <span><include src="others/inclusionInsideTable.md" /></span> | 29   | New Zealand |
</d-table>

---
Test sortable with searchable

<d-table sortable searchable>
| Name | Age | Country |
|------|-----|---------|
| John | 28  | USA     |
| Emily| 32  | Canada  |
| Michael | 41 | UK    |
| Sophia | 255 | Germany|
| David | 37  | Australia |
| <span><include src="others/inclusionInsideTable.md" /></span> | 29   | New Zealand |
</d-table>

---
Test table can be included

<span><include src="others/tableToBeIncluded.md" /></span>

---
Test table can be included in a panel

<panel type="minimal" header="This is your header for a Panel, click me to expand!">

<span><include src="others/tableToBeIncluded.md" /></span>

</panel>

---
Test table can be included in a modal

<trigger for="modal:loremipsum">trigger</trigger>
<modal header="**Modal header** :rocket:" id="modal:loremipsum">

<span><include src="others/tableToBeIncluded.md" /></span>

</modal>

---
