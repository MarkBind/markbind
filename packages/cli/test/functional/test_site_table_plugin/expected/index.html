<!DOCTYPE html>
<html>

<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="generator" content="MarkBind 5.4.0">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>Hello World</title>
  <link rel="stylesheet" href="/markbind/css/bootstrap.min.css">
  <link rel="stylesheet" href="/markbind/fontawesome/css/all.min.css">
  <link rel="stylesheet" href="/markbind/glyphicons/css/bootstrap-glyphicons.min.css">
  <link rel="stylesheet" href="/markbind/css/codeblock-dark.min.css">
  <link rel="stylesheet" href="/markbind/css/markbind.min.css">
  <script src="/markbind/js/polyfill.min.js"></script>
  <script src="/markbind/js/vue.min.js"></script>
  <script src="/markbind/js/markbind.min.js"></script>
  <script src="index.page-vue-render.js"></script>
  <link rel="stylesheet" href="/plugins/dataTable/datatables.min.css">
  <link rel="stylesheet" href="/plugins/dataTable/datatables-additional.css">
  <link rel="stylesheet" href="/plugins/markbind-plugin-anchors/markbind-plugin-anchors.css">
  <link rel="stylesheet" href="/plugins/markbind-plugin-tree/markbind-plugin-tree.css">
</head>
<script>
  const baseUrl = ''
</script>

<body>
  <div id="app">
    <header></header>
    <div id="flex-body">
      <div id="content-wrapper">
        <h1 id="welcome-to-markbind">Welcome to MarkBind<a class="fa fa-anchor" href="#welcome-to-markbind" onclick="event.stopPropagation()"></a></h1>
        <p>This is to test data tables.</p>
        <hr>
        <p>Test searchable</p>
        <div class="table-responsive">
          <table class="markbind-table table table-bordered table-striped searchable-table" id="datatable-0" v-datatable>
            <thead>
              <tr>
                <th>Name</th>
                <th>Age</th>
                <th>Country</th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>John</td>
                <td>28</td>
                <td>USA</td>
              </tr>
              <tr>
                <td>Emily</td>
                <td>32</td>
                <td>Canada</td>
              </tr>
              <tr>
                <td>Michael</td>
                <td>41</td>
                <td>UK</td>
              </tr>
              <tr>
                <td>Sophia</td>
                <td>255</td>
                <td>Germany</td>
              </tr>
              <tr>
                <td>David</td>
                <td>37</td>
                <td>Australia</td>
              </tr>
              <tr>
                <td><span>
                    <div>
                      <p><code class="hljs inline no-lang" v-pre>test</code></p>
                    </div>
                  </span></td>
                <td>29</td>
                <td>New Zealand</td>
              </tr>
            </tbody>
          </table>
        </div>
        <hr>
        <p>Test sortable</p>
        <div class="table-responsive">
          <table class="markbind-table table table-bordered table-striped sortable-table" id="datatable-1" v-datatable>
            <thead>
              <tr>
                <th>Name</th>
                <th>Age</th>
                <th>Country</th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>John</td>
                <td>28</td>
                <td>USA</td>
              </tr>
              <tr>
                <td>Emily</td>
                <td>32</td>
                <td>Canada</td>
              </tr>
              <tr>
                <td>Michael</td>
                <td>41</td>
                <td>UK</td>
              </tr>
              <tr>
                <td>Sophia</td>
                <td>255</td>
                <td>Germany</td>
              </tr>
              <tr>
                <td>David</td>
                <td>37</td>
                <td>Australia</td>
              </tr>
              <tr>
                <td><span>
                    <div>
                      <p><code class="hljs inline no-lang" v-pre>test</code></p>
                    </div>
                  </span></td>
                <td>29</td>
                <td>New Zealand</td>
              </tr>
            </tbody>
          </table>
        </div>
        <hr>
        <p>Test sortable with searchable</p>
        <div class="table-responsive">
          <table class="markbind-table table table-bordered table-striped sortable-searchable-table" id="datatable-2" v-datatable>
            <thead>
              <tr>
                <th>Name</th>
                <th>Age</th>
                <th>Country</th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>John</td>
                <td>28</td>
                <td>USA</td>
              </tr>
              <tr>
                <td>Emily</td>
                <td>32</td>
                <td>Canada</td>
              </tr>
              <tr>
                <td>Michael</td>
                <td>41</td>
                <td>UK</td>
              </tr>
              <tr>
                <td>Sophia</td>
                <td>255</td>
                <td>Germany</td>
              </tr>
              <tr>
                <td>David</td>
                <td>37</td>
                <td>Australia</td>
              </tr>
              <tr>
                <td><span>
                    <div>
                      <p><code class="hljs inline no-lang" v-pre>test</code></p>
                    </div>
                  </span></td>
                <td>29</td>
                <td>New Zealand</td>
              </tr>
            </tbody>
          </table>
        </div>
        <hr>
        <p>Test table can be included</p>
        <p><span>
            <div>
              <div class="table-responsive">
                <table class="markbind-table table table-bordered table-striped sortable-searchable-table" id="datatable-3" v-datatable>
                  <thead>
                    <tr>
                      <th>Name</th>
                      <th>Age</th>
                      <th>Country</th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr>
                      <td>John</td>
                      <td>28</td>
                      <td>USA</td>
                    </tr>
                    <tr>
                      <td>Emily</td>
                      <td>32</td>
                      <td>Canada</td>
                    </tr>
                    <tr>
                      <td>Michael</td>
                      <td>41</td>
                      <td>UK</td>
                    </tr>
                    <tr>
                      <td>Sophia</td>
                      <td>255</td>
                      <td>Germany</td>
                    </tr>
                    <tr>
                      <td>David</td>
                      <td>37</td>
                      <td>Australia</td>
                    </tr>
                    <tr>
                      <td>Emma</td>
                      <td>29</td>
                      <td>New Zealand</td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </span></p>
        <hr>
        <p>Test table can be included in a panel</p>
        <panel type="minimal"><template #header>
            <p>This is your header for a Panel, click me to expand!</p>
          </template>
          <p><span>
              <div>
                <div class="table-responsive">
                  <table class="markbind-table table table-bordered table-striped sortable-searchable-table" id="datatable-4" v-datatable>
                    <thead>
                      <tr>
                        <th>Name</th>
                        <th>Age</th>
                        <th>Country</th>
                      </tr>
                    </thead>
                    <tbody>
                      <tr>
                        <td>John</td>
                        <td>28</td>
                        <td>USA</td>
                      </tr>
                      <tr>
                        <td>Emily</td>
                        <td>32</td>
                        <td>Canada</td>
                      </tr>
                      <tr>
                        <td>Michael</td>
                        <td>41</td>
                        <td>UK</td>
                      </tr>
                      <tr>
                        <td>Sophia</td>
                        <td>255</td>
                        <td>Germany</td>
                      </tr>
                      <tr>
                        <td>David</td>
                        <td>37</td>
                        <td>Australia</td>
                      </tr>
                      <tr>
                        <td>Emma</td>
                        <td>29</td>
                        <td>New Zealand</td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </span></p>
        </panel>
        <hr>
        <p>Test table can be included in a modal</p>
        <p>
          <trigger for="modal:loremipsum">trigger</trigger>
        </p>
        <modal id="modal:loremipsum"><template #header><strong>Modal header</strong> ðŸš€</template>
          <p><span>
              <div>
                <div class="table-responsive">
                  <table class="markbind-table table table-bordered table-striped sortable-searchable-table" id="datatable-5" v-datatable>
                    <thead>
                      <tr>
                        <th>Name</th>
                        <th>Age</th>
                        <th>Country</th>
                      </tr>
                    </thead>
                    <tbody>
                      <tr>
                        <td>John</td>
                        <td>28</td>
                        <td>USA</td>
                      </tr>
                      <tr>
                        <td>Emily</td>
                        <td>32</td>
                        <td>Canada</td>
                      </tr>
                      <tr>
                        <td>Michael</td>
                        <td>41</td>
                        <td>UK</td>
                      </tr>
                      <tr>
                        <td>Sophia</td>
                        <td>255</td>
                        <td>Germany</td>
                      </tr>
                      <tr>
                        <td>David</td>
                        <td>37</td>
                        <td>Australia</td>
                      </tr>
                      <tr>
                        <td>Emma</td>
                        <td>29</td>
                        <td>New Zealand</td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </span></p>
        </modal>
        <hr>
      </div>
      <scroll-top-button></scroll-top-button>
    </div>
    <footer></footer>
  </div>
</body>
<script src="/markbind/js/bootstrap-utility.min.js"></script>
<script>
  MarkBind.setupWithSearch()
</script>
<script src="/plugins/dataTable/datatables.min.js"></script>

<script>
    function getTableOptions(el) {
      const options = {};
      if ($(el).hasClass('sortable-table')) {
        options.searching = false;
        options.paging = false;
        options.info = false;
      } else if ($(el).hasClass('searchable-table')) {
        options.ordering = false;
        options.paging = false;
        options.info = false;
        options.dom = '<"row"<"col-sm-12"f>>' + '<"row"<"col-sm-12"t>>';
      } else if ($(el).hasClass('sortable-searchable-table')) {
        options.paging = false;
        options.info = false;
        options.dom = '<"row"<"col-sm-12"f>>' + '<"row"<"col-sm-12"t>>';
      }
      return options;
    }

    Vue.directive('datatable', {
      inserted: function(el, binding) {
        const options = binding.value || {};
        const tableOptions = getTableOptions(el);
        $(el).DataTable({ ...tableOptions, ...options });
      }
    });

    document.addEventListener('DOMContentLoaded', function() {
      $('table.sortable-table, table.searchable-table, table.sortable-searchable-table').each(function() {
        const options = getTableOptions(this);
        $(this).DataTable(options);
      });
    });
  </script>

</html>